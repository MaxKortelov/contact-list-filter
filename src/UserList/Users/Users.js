import React, {useState, useEffect} from 'react';
import styles from './Users.module.css';
import {Avatar, IconButton, List, ListItem, ListItemAvatar,
        ListItemText, ListItemSecondaryAction} from '@material-ui/core/';
import DeleteIcon from '@material-ui/icons/Delete';
import EditIcon from '@material-ui/icons/Edit';
import Pagination from '@material-ui/lab/Pagination';

function Users({workList, deleteUser, editor}) {

    //Пагинация
    const [pages, setPages] = useState(1);
    const [page, setPage] = useState(10);

    const [list, setList] = useState([...workList]);
    useEffect(() => {
        setList([...workList]);
        let newPage = Math.ceil(workList.size / 10);
        setPages(newPage);
    }, [workList]);

    //Рендеринг элементов списка
    let listSet = new Set();
    list.forEach((el, id) => {
        if(page - 10 <= id && id < page) {
        listSet.add(
            <ListItem
                className={styles.user}
                key={el.unique}
            >
                <ListItemAvatar className={styles.avatar}>
                    <Avatar />
                </ListItemAvatar>
                <ListItemText
                    primary='First name:'
                    secondary={el.name}
                />
                <ListItemText
                    primary='Surname:'
                    secondary={el.surname}
                />
                <ListItemText
                    primary='Company:'
                    secondary={el.company}
                />
                <ListItemSecondaryAction className={styles.actions}>
                    <IconButton edge="end" aria-label="edit" onClick={() => editor(el)}>
                        <EditIcon />
                    </IconButton>
                    <IconButton edge="end"
                                aria-label="delete"
                                onClick={() => deleteUser(el)}
                    >
                        <DeleteIcon />
                    </IconButton>
                </ListItemSecondaryAction>
            </ListItem>
        )}
    });

    return (
        <div className={styles.main}>
        <div className={styles.pagination}>
            <Pagination count={pages}
                        color="primary"
                        onChange={(e, num) => setPage(num * 10)}
                        disabled={pages === 1}
            />
        </div>
        <div>
            <List className={styles.users}>
                {listSet}
            </List>
        </div>
        </div>
    )
}

export default Users;
