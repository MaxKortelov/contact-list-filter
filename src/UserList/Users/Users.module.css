.main {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 85vh;
    min-height: 85vh;
    width: 100%;
    padding-top: 15vh;
}

.users {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    width: 50%;
}

.user {
    border-top: 1px solid #c5c5ca;
    border-bottom: 1px solid #c5c5ca;
}

.pagination {
    margin: 1vw 0;
    height: 3vw;
    min-height: max-content;
}

.wrap {
    display: flex;
    justify-content: center;
    width: 100%;
}

@media (max-width: 880px) {
    .users {
        width: 80%;
    }
}

@media (max-width: 555px) {
    .users {
        width: 100%;
    }
}

@media (max-width: 440px) {
    .avatar {
        display: none;
    }
}

@media (max-width: 375px) {
    .avatar {
        display: none;
    }

    .actions {
        display: none;
    }
}