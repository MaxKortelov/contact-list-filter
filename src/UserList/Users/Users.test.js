import React from 'react';
import renderer from 'react-test-renderer';
import Users from './Users';

test('Users displays correctly', () => {
    let workList= [{
        name: 'Max',
        surname: 'Kortelov',
        company: '',
        unique: 123
    }];
    let deleteUser= {...workList[0]};
    let editor={...workList[0]};

    const component = renderer.create(
        <Users workList={workList}
               deleteUser={deleteUser}
               editor={editor}
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});