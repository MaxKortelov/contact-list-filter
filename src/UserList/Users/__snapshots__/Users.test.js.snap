// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users displays correctly 1`] = `
<div
  className="main"
>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root"
  >
    <ul
      className="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-selected Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          1
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
  <ul
    className="MuiList-root users MuiList-padding"
  >
    <li
      className="MuiListItem-container"
    >
      <div
        className="MuiListItem-root user MuiListItem-gutters MuiListItem-secondaryAction"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root avatar"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            First name:
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            Max
          </p>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Surname:
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            Kortelov
          </p>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Company:
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            
          </p>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root actions"
      >
        <button
          aria-label="edit"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="delete"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
        </button>
      </div>
    </li>
  </ul>
</div>
`;
